Fix static properties

BladeMF opened the pull request:
Index static properties without $

BladeMF committed:
Index static properties without the $.

BladeMF committed:
Index static properties without the $.

BladeMF committed:
Add static property indexer test

BladeMF committed:
Add static property indexer test

dantleech commented on tests/Adapter/ReferenceFinder/IndexedReferenceFinderTest.php:
```diff
@@ -45,6 +45,7 @@ public function testFinder(string $manifest, int $expectedConfirmed, ?int $expec
 
         $locations = iterator_to_array($locations);
 
+        // dump($locations);
```
dump

dantleech commented:
25LOC - not bad :+1: Not sure why the CI didn't start, will have to check that, there is a stray `dump`

BladeMF committed:
Remove stray dump

BladeMF committed:
Remove stray dump

BladeMF commented:
> 25LOC

What is LOC?

BladeMF commented:
The unit test fails because it depends on WorseReflection working properly.

BladeMF commented:
https://github.com/phpactor/phpactor/issues/1154

dantleech commented:
> **What is LOC?**

Lines Of Code

BladeMF commented:
> > **What is LOC?**
> 
> Lines Of Code

Your code is pretty well structured - I don't need to do much.

BladeMF committed:
Index use statements and trait usings.

BladeMF committed:
Index use statements and trait usings.

BladeMF committed:
-- Index trait usings as references
-- Added tests for trait usings for IndexedReferenceFinder
-- Added traits for TraitUseCaluseIndexer

BladeMF committed:
-- Index trait usings as references
-- Added tests for trait usings for IndexedReferenceFinder
-- Added traits for TraitUseCaluseIndexer

BladeMF committed:
Coding standards

BladeMF committed:
Coding standards

BladeMF commented:
What do we need to do to tell it to use the new version of worse-reflection?

dantleech commented:
Typically we reference the _next_ version in `composer.json`. So if WR is tagged as `0.1.4` then we require `0.1.5`. Because `minimum-stability: dev` is used it will pick up dev-master (while we use `prefer-stable` to ensure that we don't get dev versions of all packages).

You will need to require `worse-reflection @ 0.4.4`

BladeMF commented:
Currently is `"phpactor/worse-reflection": "^0.4",`. Shouldn't it be 0.5?

BladeMF committed:
change worse-reflection version requirement

BladeMF committed:
change worse-reflection version requirement

BladeMF commented:
What should we do about 7.4 test failing?

dantleech commented:
It's the same flakey test I think, I restarted it and it's green now (should increasing the `delay()` time would probably be a very quick fix)

dantleech commented on composer.json:
```diff
@@ -20,7 +20,7 @@
         "phpactor/source-code-filesystem-extension": "^0.1.3",
         "phpactor/text-document": "^1.1.3",
-        "phpactor/worse-reflection": "^0.4",
+        "phpactor/worse-reflection": "^0.4.4",
```
:+1:

dantleech commented on lib/Adapter/Tolerant/Indexer/ClassLikeReferenceIndexer.php:
```diff
@@ -48,10 +50,13 @@ public function beforeParse(Index $index, SplFileInfo $info): void
         assert($node instanceof QualifiedName);
+        
+        $name =
+            ($node->parent->parent instanceof TraitUseClause) ?
```
don't need the parenthesis here

dantleech commented on tests/Adapter/ReferenceFinder/IndexedReferenceFinderTest.php:
```diff
@@ -102,6 +103,54 @@ class Bar extends Foo {}
+// <?php
+// $b = new Foo();
+// $b->func();
+// EOT
```
please remove all commented code (in general no TODOs, but especially not commented code).

In general this stuff just doesn't age well.

BladeMF replied:
Ye, I keep forgetting those. Sorry about that.

BladeMF replied:
Actually I didn't forget it - I meant to leave it as a failing test should someone decide to go after it.

dantleech commented:
very nice :+1: just a few comments

BladeMF committed:
fix comments

BladeMF committed:
fix comments

BladeMF committed:
Merge pull request #1 from phpactor/master

Merge

BladeMF committed:
Merge branch 'fix-static-properties' of github.com:BladeMF/indexer-extension into fix-static-properties

BladeMF commented:
Rebased on master

dantleech commented:
Great work :tada:
